src/shortcut.js

```diff
@@ -0,0 +1,330 @@
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress() &&
+      selected && selected.isDeletable() && selected.isMovable() &&
+          !selected.workspace.isFlyout;
+    },
+    callback: function() {
+      copyData.clear();
+      const apply = function(block) {
+        if (block &&
+            block.isDeletable() &&
+            !hasSelectedParent(block)) {
+          copyData.add(block.toCopyData());
+          if (block.workspace.isFlyout) {
+            return;
+          }
+          block.workspace.hideChaff();
+          if (block.outputConnection) {
+            block.dispose(false, true);
+          } else {
+            block.dispose(true, true);
+          }
+        }
+      };
+      const selected = Blockly.common.getSelected();
+      Blockly.Events.setGroup(true);
+      if (!blockSelection.size) {
+        apply(selected);
+      }
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        apply(block);
+      });
+      Blockly.Events.setGroup(false);
+      return true;
+    },
+  };
+
+  Blockly.ShortcutRegistry.registry.register(cutShortcut);
+
+  const ctrlX = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.CTRL]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(ctrlX, cutShortcut.name);
+
+  const altX =
+  Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.ALT]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(altX, cutShortcut.name);
+
+  const metaX = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.META]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(metaX, cutShortcut.name);
+};
+
+/**
+ * Keyboard shortcut to paste multiple selected blocks on
+ * ctrl+v, cmd+v, or alt+v.
+ */
+const registerPaste = function() {
+  const pasteShortcut = {
+    name: Blockly.ShortcutItems.names.PASTE,
+    preconditionFn: function(workspace) {
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress();
+    },
+    callback: function() {
+      Blockly.Events.setGroup(true);
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (block) {
+          block.pathObject.updateSelected(false);
+        }
+      });
+      blockSelection.clear();
+      copyData.forEach(function(data) {
+        if (!data) {
+          return null;
+        }
+        // Pasting always pastes to the main workspace, even if the copy
+        // started in a flyout workspace.
+        let workspace = data.source;
+        if (workspace.isFlyout) {
+          workspace = workspace.targetWorkspace;
+        }
+        if (data.typeCounts &&
+            workspace.isCapacityAvailable(data.typeCounts)) {
+          const block = workspace.paste(data.saveInfo);
```

Comment:
You should be able to use `clipboard.paste` instead.

Reply:
The reason I choose to re-realize the clipboard.paste https://github.com/google/blockly/blob/61322294da70c770b32abb28b9d695af75f61a16/core/clipboard.js#L46-L61 is that, I can't get access to the `copyData` directly
https://github.com/google/blockly/blob/61322294da70c770b32abb28b9d695af75f61a16/core/clipboard.js#L26 It will simplify my code a lot and reducing a lot of duplicating work if that is accessible through API.

Reply:
Mmm you are correct. Will file a bug!

Reply:
Filed: https://github.com/google/blockly/issues/6275